omnigibson/utils/python_utils.py

```diff
@@ -709,6 +709,76 @@ def deserialize(cls, state):
         return state_dict
 
 
+class CachedFunctions:
```

Comment:
oh good point -- yeah, this is definitely more ideal! A few problems though:

- There's some cases where you'd want the control dict to be different but _not_ the underlying function to be changed. Case in point: Tiago's right hand eef position. It's shifted by the trunk value for the sake of control, but we definitely should not be overloading the underlying `get_eef_pose` call.
- The control dict depends on the robot type. A manipulation robot will have more entries than a locomotion-only robot. We could have type-checking for each call, but that seems a bit ugly
- In manipulation controllers, it's not clear how it will know what arm it will need to query. I suppose we can pass in the "arm" name as a kwarg but that also seems a bit ugly

Reply:
ok sg lets do this